File: 300.600/344.ReverseString.cpp
#include <iostream>
#include <vector>
#include <utility>

using namespace std;

class ReverseString {
public:
    void reverseString(vector<char>& s) {
        int b = 0, e = s.size() - 1;
        while (b < e) {
            swap(s[b], s[e]);
            ++b;
            --e;
        }
    }
};

int main(int argc, char const* argv[]) {
    ReverseString cls;
    vector<char> vec{ 'h','e','l','l','o' };
    cls.reverseString(vec);
    cout << "[";
    for (auto c : vec) {
        cout << c << ",";
    }
    cout << "]" << endl;

    vector<char> vec1{ 'H','a','n','n','a','h' };
    cls.reverseString(vec1);
    cout << "[";
    for (auto c : vec1) {
        cout << c << ",";
    }
    cout << "]" << endl;

    return 0;
}

File: go/300.500/344.ReverseString.go
package main

import "fmt"

func reverseString(s []byte) {
	b, e := 0, len(s)-1
	for b < e {
		s[b], s[e] = s[e], s[b]
		b++
		e--
	}
}

func main() {
	s := []byte{'h', 'e', 'l', 'l', 'o'}
	reverseString(s)
	fmt.Println(string(s))

	s = []byte{'H', 'a', 'n', 'n', 'a', 'h'}
	reverseString(s)
	fmt.Println(string(s))
}
